sdk/lib/opentelemetry/sdk/configurator.rb

```diff
@@ -181,7 +184,7 @@ def wrapped_exporters_from_env # rubocop:disable Metrics/CyclomaticComplexity, M
               OpenTelemetry.logger.warn "The #{otlp_protocol} transport protocol is not supported by the OTLP exporter, spans will not be exported."
               nil
             else
-              fetch_exporter(exporter, 'OpenTelemetry::Exporter::OTLP::Exporter')
+              fetch_exporter(exporter, 'OpenTelemetry::Exporter::OTLP::Exporter', metrics_reporter, metrics_reporter: metrics_reporter)
```

Comment:
This stuttering feels awkward. There's not a great alternative, though (e.g. just passing the hash and fishing out the `:metrics_reporter` from it is its own brand of awkward). I think we should make the hash explicit, though - I was confused initially thinking this was a keyword argument.

```suggestion
              fetch_exporter(exporter, 'OpenTelemetry::Exporter::OTLP::Exporter', metrics_reporter, { metrics_reporter: metrics_reporter })
```

Reply:
It does indeed feel awkward. 

How do you feel about extracting BSP arguments into its own keyword arg and making `exporter_options` a keyword?

E.g.: 
```ruby
def fetch_exporter(name, class_name, processor_options: {}, exporter_options: {})
```

Reply:
I mean, it'll be even more stuttering, but I think it'd be a lot clearer:
```suggestion
              fetch_exporter(exporter, 'OpenTelemetry::Exporter::OTLP::Exporter',
                             processor_options: { metrics_reporter: metrics_reporter },
                             exporter_options: { metrics_reporter: metrics_reporter })
```

Reply:
👍🏼

Reply:
Another option may be to separate span processor instantiation from exporter initialization. Make `fetch_exporter` return an `exporter` instead of a BSP and then instantiate the span processor based on the Exporter type.

Thoughts?

Reply:
That looks like:
```ruby
          exp = case exporter.strip
          when 'none' then nil
          when 'otlp'
            otlp_protocol = ENV['OTEL_EXPORTER_OTLP_TRACES_PROTOCOL'] || ENV['OTEL_EXPORTER_OTLP_PROTOCOL'] || 'http/protobuf'
            if otlp_protocol != 'http/protobuf'
              OpenTelemetry.logger.warn "The #{otlp_protocol} transport protocol is not supported by the OTLP exporter, spans will not be exported."
              nil
            else
              fetch_exporter(exporter, 'OpenTelemetry::Exporter::OTLP::Exporter', { metrics_reporter: metrics_reporter })
            end
          when 'jaeger' then fetch_exporter(exporter, 'OpenTelemetry::Exporter::Jaeger::CollectorExporter')
          when 'zipkin' then fetch_exporter(exporter, 'OpenTelemetry::Exporter::Zipkin::Exporter')
          when 'console' then next Trace::Export::SimpleSpanProcessor.new(Trace::Export::ConsoleSpanExporter.new)
          else
            OpenTelemetry.logger.warn "The #{exporter} exporter is unknown and cannot be configured, spans will not be exported"
            nil
          end
          Trace::Export::BatchSpanProcessor.new(exp, metrics_reporter: metrics_reporter) unless exp.nil?
```
This seems reasonable to me. Note that we need the `next` for the `console` exporter, since that's wrapped in a SSP instead of a BSP.

Reply:
I started experimenting with a few different implementations but the asymmetry of the Jaeger and Zipkin exporters is bugging me so I am going to add the reporter to those exporters as well in a separate PR and then circle back to this.

Reply:
Now that the jaeger and zipkin exporters accept a metrics reporter the code is symmetric and should simplify this a little.